src/app/store/settings/settings.reducer.ts

```diff
@@ -20,6 +20,7 @@ export interface SettingsState {
   researchSpeed: ResearchSpeed;
   flowRate: number;
   expensive: boolean;
+  theme?: Theme;
```

Comment:
I don't think I would declare this as optional with the `?`, is there a reason for it?

Reply:
Two reasons:

1. I didn't like making a bunch of changes to the router service to set the theme in the url data.
2. I intended (a patch is coming later because it doesn't work at the moment) to have the default value be unset which would fall back to the web browser's chosen [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) value.

Reply:
That's fair, if it might be null to fall back to default browser setting, this is how it should be.

Reply:
I got the OS default working fine and in Windows 10 you can switch between light and dark mode now by opening your color settings and switching Windows to dark mode. I am not sure I like this particular change because most users would likely have light mode by default...

Also when you open the settings there is nothing selected in this option now.

If you think it should default to selected dark-mode (overriding OS preference) let me know and I can improve the feature.

Reply:
I think it'd be better to keep the default as dark mode regardless of os setting. TBH, I'm not even sure I've noticed dark mode for Windows myself and will now have to look for it. 🙂 That's a really cool idea in theory but probably doesn't work well in this case.

Reply:
Sounds good, I dropped the media preference fallbacks from values.scss (which could maybe be renamed theme.scss now?).

`theme` is still marked optional because it is not being persisted in the route query data. As I see it alternatives are:

1. Persist theme to route data and update tests and the service.
2. Introduce a new type for the theme related state.

Reply:
I don't think theme should be persisted to the route data, if you're sharing the link with someone else you don't want to override their theme preferences. Ultimately this preference should be saved to Local Storage instead so it's preserved for the current browser, but that can be addressed later. 

One last thing I wanted to point out: Even though the theme is not persisted to the route, it should never be null. When settings are restored for the route, they override any settings that are defined, but for settings that are not defined they are left at the default - what is defined in `initialSettingsState`. For this reason I don't think theme would ever be null, and I think it'd be safe to take out the `?` flag on the field definition. Other than that I'm ready to merge this!

Reply:
How should the route tests be changed to permit the `?` to be removed?

Reply:
I see what you mean. Seems like it should be trivial to handle but not entirely true. Seems better to leave the `?` after all.